services/network/networksecuritygroup/wssd.go

```diff
@@ -271,44 +271,28 @@ func getWssdNetworkSecurityGroupRules(securityRules *[]network.SecurityRule, isD
 			return nil, errors.Wrapf(errors.InvalidInput, "Unknown Protocol %s specified", rule.Protocol)
 		}
 
-		if rule.SourceAddressPrefix != nil {
+		if rule.SourceAddressPrefix != nil && len(*rule.SourceAddressPrefix) > 0 {
 			wssdCloudNSGRule.SourceAddressPrefix = *rule.SourceAddressPrefix
 		} else if rule.SourceAddressPrefixes != nil {
-			concatRule := ""
-			for _, prefix := range *rule.SourceAddressPrefixes {
-				concatRule += prefix
-			}
-			wssdCloudNSGRule.SourceAddressPrefix = concatRule
+			wssdCloudNSGRule.SourceAddressPrefix = strings.Join(*rule.SourceAddressPrefixes, ",")
```

Comment:
>strings.Join(*rule.SourceAddressPrefixes, ",") [](http://example.com/codeflow?start=42&length=46)

i remember you mentioned operators did some workarounds because of this bug, will those workarounds continue to work once we fix it? or do we need operators to change?

Reply:
Yes, I'll reach out to them to remove those fixes! I believe their fix was to concatenate the arrays themselves and to pass an array with one element that is a concatenated string, so that would still work. I can make sure to notify Arka and whoever else needs to know when this is merged.

Reply:
ok, as this is low pri fix, let's hold on and dont merge until we are 100% sure. i would recommend sharing a private with them and testing. for context, our asz private preview is going to happen in 2408 we need to be extra careful with anything we merge.